pycbc/filter/matchedfilter.py

```diff
@@ -1610,8 +1613,11 @@ def _process_vetoes(self, results, veto_info):
         """Calculate signal based vetoes"""
         chisq = numpy.array(numpy.zeros(len(veto_info)), numpy.float32, ndmin=1)
         dof = numpy.array(numpy.zeros(len(veto_info)), numpy.uint32, ndmin=1)
+        sg_chisq = numpy.array(numpy.zeros(len(veto_info)), numpy.float32,
```

Comment:
I think this should be numpy.ones as the default for sgchisq.

Reply:
Not sure I agree. A value of 0 is consistent with "This veto is not being calculated", whereas 1 is more consistent with "The veto is on, but not calculated for this trigger" (or am I misunderstanding).

Reply:
You somehow got in between me resolving this comment. It is fine because sg_chisq fills in the ones for you. The problem would occur down the line, as the ranking statistic doesn't sanity check this sort of thing. Because of how this is normally applied it would fail with a zero value (unlike bruce chisq for which newsnr is perfectly well defined for a value of zero).

Reply:
In any case, it is fine to leave this as zeros.

Reply:
Well new snr is also ill-defined when chi-squared DOF is also 0 ... I think that's the *right* behaviour. The code should fail if computing a statistic that you don't have information for, rather than proceeding happily.

Reply:
The point is the statistic has decided what it wants to do when it isn't run for some reason, and that is to assign a value that won't change the ranking. That would be 1, and not 0. It will never output a zero as currently run.